python/lsst/obs/base/ingest_tests.py

```diff
@@ -109,33 +106,55 @@ def instrumentName(self):
         """
         return self.instrumentClass.getName()
 
-    def setUp(self):
+    @classmethod
+    def setUpClass(cls):
         # Use a temporary working directory
-        self.root = tempfile.mkdtemp(dir=self.ingestDir)
-        self._createRepo()
+        cls.root = tempfile.mkdtemp(dir=cls.ingestDir)
+        cls._createRepo()
 
         # Register the instrument and its static metadata
-        self._registerInstrument()
+        cls._registerInstrument()
 
-    def tearDown(self):
-        if os.path.exists(self.root):
-            shutil.rmtree(self.root, ignore_errors=True)
+    def setUp(self):
+        # Want a unique run name per test
+        self.outputRun = "raw_ingest_" + str(random.randrange(1_000_000_000))
```

Comment:
Could you use `self.id()` here instead of a random number? That will make it obvious what test a given output directory goes with, and also prevent (rare) collisions.

Reply:
Thanks. I did not know about that. Now I wonder if utils `getTempFilePath` is trying too hard.